src/plots/plots.js

```diff
@@ -2675,9 +2675,13 @@ plots.doCalcdata = function(gd, traces) {
     Registry.getComponentMethod('errorbars', 'calc')(gd);
 };
 
-function clearAxesCalc(axList) {
+function setupAxisCategories(axList, fullData) {
     for(var i = 0; i < axList.length; i++) {
-        axList[i].clearCalc();
+        var ax = axList[i];
+        ax.clearCalc();
+        if(ax.type === 'multicategory') {
+            ax.setupMultiCategory(fullData);
+        }
```

Comment:
`clearCalc` is also used in `rangeslider.draw` and `gl2d/scene2d` - do these need `setupMultiCategory`? Should that just be part of `clearCalc` for multicategory axes?

Reply:
> `clearCalc` is also used in `rangeslider.draw` and `gl2d/scene2d` - do these need `setupMultiCategory`?

`setupMultiCategory` only needs to be called once (here in `doCalcdata`, to fill in `ax._muticatList`), so rangeslider don't need it, that's why I didn't put the `setupMultiCategory` logic in `clearCalc`. As for `gl2d/scene2d`, gl2d subplots don't support `multicategory` axes, so I think we're :ok_hand: